vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -137,13 +143,28 @@ private VimeoClient(@NotNull Configuration configuration) {
         setVimeoAccount(vimeoAccount);
     }
 
+    /**
+     * Sets a new base URL to be used for requests
+     * by the VimeoClient for specific paths. If a null
+     * base URL is provided, the default base URL will be
+     * used by the client.
+     *
+     * @param path    the path for which to change the base URL.
+     * @param baseUrl the new base URL to use, may be null.
+     */
+    public void setBaseUrlForRequest(@NotNull String path, @Nullable String baseUrl) {
+        mBaseUrlInterceptor.setBaseUrlForRequest(path, baseUrl);
+    }
```

Comment:
Another cool feature which I think we should add as part of this is is the ability set the baseUrl for all requests without needing to reinitialize. The Vimeo flagship app would get some benefit out of this with an example from our Admin panel. This also makes it more consistent from a consumer's standpoint since they won't have to use two separate techniques for baseUrl modification. It doesn't cover cert pinning disabling but we could maybe do that through `ignoreCertificates()`?

```java
Configuration configuration = VimeoClient.getInstance().getConfiguration();
configuration.setBaseURLString(newUrl);
configuration.setCertPinningEnabled(false); // disable cert pinning or endpoints will most likely fail
VimeoClient.initialize(configuration);
```

Another feature which I don't think needs to be done here is a blacklist of urls for the new url. So we hit every request with the new URL _except_ a select few. Might help get around that pattern matching issue a little.